middle_end/flambda/types/env/aliases.ml

```diff
@@ -43,6 +43,8 @@ module Map_to_canonical = struct
           if Coercion.equal coercion1 coercion2 then
             Some coercion1
           else
+            (* CR lmaurer: We might not want to die here!! Finding best
```

Comment:
Could you elaborate on what you mean? (Not necessarily to
update the comment, mainly to help me understand what
kind of change you would like to see.)

Reply:
I would also like to understand how urgent this is to fix.

Reply:
In general, it's not impossible that there are different coercions on the two sides of a join, though it should be rare. In which case we may find ourselves looking for a common ancestor among aliases with inconsistent coercions, which would cause an error here.

Reply:
I see; I was still picturing a (dated) version of coercions where
one can always compose two into a simple one, but as you say
in general this does not hold and the compositions on both
sides might be equivalent yet not equal.

Reply:
You wouldn't want to compose them, since there's no guarantee that they compose (and which way would you compose them?). Actually, though, in this case, it's safe to just drop the element, since the point of taking an intersection is to find common aliases, and it's a safe approximation to drop a common alias (at worst the join becomes less precise). I'll change it to return `None` instead of throwing an error.

Reply:
Sorry, I did not mean to compose the coercions you
get at this point. I meant that on both sides you may
have a "chain" of coercions that do not compose (as
in "simplify") to a "simple" one. However, I realize that
does not matter as much as I thought it would.

Reply:
Oh! Of course. Yeah, I can see it getting very complicated to compare coercions, so it's lucky we can be conservative here